### PR TITLE
Add --skip-observability-permissions to setup all

### PR DESCRIPTION
**Pull request by DheerajPannala**
## Summary

Adds an opt-in `a365 setup all --skip-observability-permissions` for blueprint agents that export telemetry through the app-only S2S endpoint, and fixes `setup all --agent-registration-only` exiting 0 when registration fails.

## Why

With microsoft/Agent365-nodejs#290 and microsoft/Agent365-Samples#339, agents export telemetry with an app-only token to the S2S endpoint (`/observabilityService/tenants/{tenantId}/otlp/agents/{agentId}/traces`). That endpoint authorizes registered agent instances without the `Agent365.Observability.OtelWrite` role: an end-to-end run with a roleless app-only token from a DevTools-registered agent returned HTTP 200 and the spans arrived. Today `setup all` always requests OtelWrite as both a delegated scope and an app role, and both need admin consent. This flag lets those agents skip them.

Defaults are unchanged, because the published SDKs still export to the non-S2S endpoint by default, and that endpoint does need OtelWrite.

## Changes

- `--skip-observability-permissions` removes Observability API from:
  - the permission specs (inheritable permissions, app role grants, batch consent);
  - the per-resource and combined admin consent URLs;
  - the dry-run plan and the summary's portal walkthrough.
- The flag fails fast (exit 1) for AI Teammate agents. It also fails with authMode `s2s`/`both`, whether set by flag or in `a365.config.json`, because OtelWrite is the only app role those modes grant. A contradicting `--authmode` flag is rejected before bootstrap signs in or writes config.
- With the flag, a failed agent registration is recorded as an error (exit 1), because registration is then the agent's only Observability authorization.
- Fix: `--agent-registration-only` now exits 1 when registration fails. The summary already pointed to "see errors", but the failure was recorded as a warning.
- Updated the CHANGELOG, `Commands/SetupSubcommands/README.md`, and the guided-setup observability doc.

The flag does not revoke permissions granted by earlier runs. Stale-permission cleanup already protects Observability API.

## Testing

- 15 new test cases cover spec and consent-URL wiring, validation and fail-fast order, registration severity, the dry run, and the summary. Each was confirmed to fail against a mutation of the code it guards.
- Full suite: 2027 passed, 12 skipped (pre-existing), 0 failed. The Release build has 0 warnings, and restore ran with NuGet audit enabled.
- I packed and installed the CLI (`1.1.223+bacea97843`), then checked `setup all --help`, the dry run, and the rejected combinations. The rejected combinations write no files.
- Not run live: a full `setup all --skip-observability-permissions` against a tenant, because it creates Entra objects. Roleless S2S access was verified on the `/otlp` route above; the legacy S2S route and sovereign clouds were not verified.

## Out of scope / follow-ups

- AI Teammate setup (`create-instance`) still grants OtelWrite.
- `a365 setup permissions bot` still configures Observability API.
- Consider making the skip the default once the SDKs default to the S2S endpoint.

### CODE REVIEW

File: src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/NonDwBlueprintSetupOrchestrator.cs [hunk 1]
```diff
@@ -549,15 +555,23 @@ private static async Task ExecuteAgentIdentityAndRegistrationAsync(
         ctx.Logger.LogInformation("");
         ctx.Logger.LogInformation("Registering agent...");
 
+        // Registration is the sole purpose of --agent-registration-only and, with OtelWrite skipped,
+        // the agent's only Observability authorization, so its failure must fail setup.
+        var registrationRequired = skipIdentityAndPermissions || ctx.SkipObservabilityPermissions;
```
Comment (rbrighenti) [inline src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/NonDwBlueprintSetupOrchestrator.cs]: `registrationRequired` covers the create-failure path, but the inconclusive-verification branch below (`AgentRegistrationExistsAsync` returns null, "retaining stored value") still sets `registrationAlreadyExisted = true` and setup exits 0. With `--skip-observability-permissions` registration is the agent's only authorization, so an auth or transient failure there should be an error rather than a pass. Please treat the null case as an error when `registrationRequired` is true, and add a test where the check returns null.

File: src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/SetupHelpers.cs [hunk 2]
```diff
@@ -1425,12 +1443,13 @@ internal static void ApplyConsentUrlsIfNeeded(
         if (ctx.Results.TenantWideConsentOutcome == Models.GrantOutcome.Granted || string.IsNullOrWhiteSpace(ctx.Config.AgentBlueprintId))
             return;
 
-        var consentResourceNames = PopulateAdminConsentUrls(ctx.Config, mcpResourceAppId, mcpScopes, isM365, mcpScopesByAudience, mcpAudienceDisplayNames);
+        var includeObservability = !ctx.SkipObservabilityPermissions;
+        var consentResourceNames = PopulateAdminConsentUrls(ctx.Config, mcpResourceAppId, mcpScopes, isM365, mcpScopesByAudience, mcpAudienceDisplayNames, includeObservability);
```
Comment (rbrighenti) [inline src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/SetupHelpers.cs]: `PopulateAdminConsentUrls` only upserts into `ResourceConsents`, so if an earlier run without the flag saved an Observability consent entry, it survives this run and is still surfaced to the admin. When `includeObservability` is false the Observability entry should be removed. A test that seeds the generated config with an existing Observability consent would cover it.

**Review by rbrighenti — CHANGES_REQUESTED** [review]
Requesting changes. The flag is well scoped and the incompatible-combination guards look right, but two gaps undercut the contract the docs promise ("setup exits with code 1 if registration fails"). Details inline. Both need a regression test.

File: src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/NonDwBlueprintSetupOrchestrator.cs (hunk 1 above)
Comment (DheerajPannala) [reply src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/NonDwBlueprintSetupOrchestrator.cs]: Fixed in 9e72b1b. When registration is required (`--agent-registration-only`, or Observability permissions not requested, which is now every blueprint agent), a null from `AgentRegistrationExistsAsync` is recorded as an error and setup exits 1. The stored ID is kept and no duplicate registration is created. The optional path still retains the ID as before. Regression tests: `Step6_RegistrationRequired_FailsWithoutReRegistering_WhenVerificationIsInconclusive` covers both required modes, and the existing retain test now runs on the optional path with unchanged assertions.

File: src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/SetupHelpers.cs (hunk 2 above)
Comment (DheerajPannala) [reply src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/SetupHelpers.cs]: Fixed in 9e72b1b. When `includeObservability` is false, `PopulateAdminConsentUrls` now removes an Observability `ResourceConsents` entry saved by an earlier run. Regression test: `PopulateAdminConsentUrls_WithoutObservability_RemovesObservabilityEntryFromEarlierRun` seeds a stale entry and checks that the other resources are unaffected.

File: src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/AllSubcommand.cs [hunk 3]
```diff
@@ -397,13 +397,18 @@ effectiveAuthModeForValidation is not ("obo" or "s2s" or "both"))
                 return;
             }
 
+            // Registered blueprint agents export telemetry app-only over S2S, so only the app-role modes
+            // (s2s/both) still request OtelWrite; AI Teammate setup is unchanged.
+            var skipObservabilityPermissions = nonDwConfig is not null
+                && effectiveAuthModeForValidation is not ("s2s" or "both");
```
[the remark this answers is not in the document]
Comment (DheerajPannala) [reply src/Microsoft.Agents.A365.DevTools.Cli/Commands/SetupSubcommands/AllSubcommand.cs]: Fixed in 9e72b1b. The skip now requires an effective blueprint selection (`--aiteammate false` or `IsBlueprintAgent`), so a dry run with an AI Teammate config keeps Observability. Test: `SetupAll_AiTeammateConfig_DryRunWithoutFlag_DoesNotSkipObservabilityApi`.

File: docs/agent365-guided-setup/a365-observability-instructions.md [hunk 4]
```diff
@@ -775,6 +775,8 @@ This skill is safe to rerun. On subsequent runs:
 
 ### OtelWrite App Role Assignment
 
+> **Blueprint agents (default):** `a365 setup all` does not request Observability API permissions for blueprint agents in the default auth mode — the S2S endpoint authorizes registered agent instances without the `OtelWrite` role, so no admin consent is needed. Setup exits with code 1 if registration fails; retry with `a365 setup all --agent-registration-only`. Use `--authmode both` if the agent still exports through the delegated (OBO) route. Permissions granted by earlier runs are not revoked.
+
 `a365 setup all` **attempts** to grant `Agent365.Observability.OtelWrite` to the Agent Identity SP, but this requires **Global Administrator** privileges. If the logged-in user is not a Global Admin, the assignment silently fails with 403 and trace exports will return HTTP 403 from the observability service.
 
 **The CLI prints a PowerShell admin consent script** in its output when the assignment fails. When running `a365 setup all`, **always scan the output for this script block** and display it to the user in a fenced code block so they can copy it and hand it to a Global Admin.
```
[the remark this answers is not in the document]
Comment (DheerajPannala) [reply docs/agent365-guided-setup/a365-observability-instructions.md]: Fixed in 9e72b1b. The grant steps are now scoped to AI Teammate agents and to blueprint agents whose SDK still exports through the delegated (OBO) route.

**Comment by DheerajPannala** [comment]
Review feedback addressed in 9e72b1b:

- **Registration (Rick):** an inconclusive registration check now fails setup when registration is required, keeping the stored ID and not re-registering.
- **Consent (Rick):** a stale Observability consent entry from an earlier run is removed when Observability isn't requested.
- **AI Teammate dry run (Copilot):** Observability is still requested for AI Teammate configs, including those kept for a dry run.
- **Docs (Copilot):** the guided-setup grant steps are scoped to AI Teammates and older delegated-route SDKs. The two stale doc comments are fixed.

Each fix has a regression test that fails against a mutation of the fix. Full suite: 2030 passed, 12 skipped (pre-existing), 0 failed.

File: src/Tests/Microsoft.Agents.A365.DevTools.Cli.Tests/Commands/SetupCommandTests.cs [hunk 5]
```diff
@@ -684,4 +684,122 @@ public async Task SetupAll_NoAuthMode_DefaultsToOboBehaviour()
...
+    {
+        var config = new Agent365Config
+        {
+            TenantId = "tenant",
+            AgentIdentityDisplayName = "agent",
+            AgentBlueprintDisplayName = "TestBlueprint",
+            DeploymentProjectPath = ".",
+            AiTeammate = false,
+            UseBlueprint = true,
+            AuthMode = configAuthMode,
```
[the remark this answers is not in the document]
Comment (DheerajPannala) [reply src/Tests/Microsoft.Agents.A365.DevTools.Cli.Tests/Commands/SetupCommandTests.cs]: The fixture already sets `AiTeammate = false` (line 733), and the command passes `--aiteammate false` (line 740), so `aiTeammateFlag == false` selects the blueprint path. All three cases pass locally and in CI on 9e72b1b.

File: CHANGELOG.md [hunk 6]
```diff
@@ -22,6 +22,8 @@ Agents provisioned before this release need `Agent365.Observability.OtelWrite` g
 
 **Option B — CLI** (`a365 setup admin`) has been removed in this release. Use Option A above, or copy the PowerShell instructions printed in the `a365 setup all` summary output.
 
+Blueprint agents that export telemetry through the app-only S2S endpoint do not need these permissions: `a365 setup all` no longer requests them for blueprint agents, and agent registration authorizes the agent instead. Grant them with Option A only for agents that still export through the delegated (OBO) route (#501).
```
[the remark this answers is not in the document]
Comment (DheerajPannala) [reply CHANGELOG.md]: Condensed in cd937db to one consumer-facing sentence. I also updated the related Fixed entry: it mentioned only `--agent-registration-only`, but setup now also exits 1 for blueprint agents when registration fails or cannot be verified.

File: CHANGELOG.md [hunk 7]
```diff
@@ -22,6 +22,8 @@ Agents provisioned before this release need `Agent365.Observability.OtelWrite` g
 
 **Option B — CLI** (`a365 setup admin`) has been removed in this release. Use Option A above, or copy the PowerShell instructions printed in the `a365 setup all` summary output.
 
+Blueprint agents that export telemetry through the app-only S2S endpoint no longer need these permissions; grant them with Option A only for agents that still export through the delegated (OBO) route (#501).
```
[the remark this answers is not in the document]
Comment (DheerajPannala) [reply CHANGELOG.md]: Fixed in 3d54061: the heading and opening requirement now apply to agents that export through the delegated (OBO) route, and the closing sentence covers blueprint agents on the S2S endpoint, so the note gives one consistent instruction.
